src/kafka-util/src/bin/kgen.rs

```diff
@@ -584,16 +617,26 @@ async fn main() -> anyhow::Result<()> {
             eprintln!("Generating message {}", i);
         }
         value_gen.next_value(&mut buf);
-        let key_i = if let Some(key_dist) = key_dist.as_ref() {
-            key_dist.sample(&mut rng)
+        let key = if let Some(key_gen) = key_gen.as_mut() {
+            // TODO: find a nicer way of doing this. The API is designed to
+            // reuse the same buffer over and over again for random data but the
+            // alternative key generation methods currently want to allocate their
+            // a new buffer each time so for now, we're going with a mildly silly
+            // workaround.
```

Comment:
Does something like this not work?

```rust
let mut key_buf = vec![];
for i in 0..n {
    key_buf.clear();
    if let Some(key_gen) = &mut key_gen {
         key_gen.next_value(&mut key_buf);
    } else if let Some(key_dist) = &key_dist {
        key_buf.extend(key_dist.sample(&mut rng).to_be_bytes());
    } else {
        key_buf.extend(i.to_be_bytes())
    };
    // ...
}
```

Reply:
not quite, you need to call `.iter()` on the arrays. But in principle it would, yeah, good catch!

I'm a bit embarrassed that I missed this obvious thing when proposing my convoluted solution. The Rust stdlib's surface area is hard to keep in my head!